### DISCUSSION TITLE
barcodes associated to calculated clusters

### DISCUSSION THREAD

**Discussion by johaGL** [theislab/scvelo#522] [Q&A] [answered]
Dear theislab, 
Thank you for your great software. I ran scvelo obtaining 14 clusters by Louvain algorithm. I would like to obtain the list of the barcodes and their associated cluster, could you please indicate me how to obtain this?

**Comment by WeilerP** [comment, marked as answer]
The info you are looking for is stored in `adata.obs['louvain']` by default, unless you specified `key_added` when calling `scv.tl.louvain`. See e.g. [this](https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html) Scanpy tutorial on clustering.

**Reply by johaGL** [reply]
thank you very much ! It works.
